File: benchmarks/cross-language/string_ops/bench.cs
using System;

class Program
{
    static void Main()
    {
        // String concatenation and manipulation
        string result = "";
        for (int i = 0; i < 10000; i++)
        {
            result = result + i.ToString();
        }

        // String methods
        long count = 0;
        for (int j = 0; j < 1000; j++)
        {
            string upper = result.ToUpper();
            string lower = result.ToLower();
            string[] parts = result.Split('5');
            count += parts.Length;
        }
        Console.WriteLine($"operations: {count}");
    }
}

File: benchmarks/cross-language/string_ops/bench.py
def main():
    # String concatenation and manipulation
    result = ""
    for i in range(10000):
        result = result + str(i)

    # String methods
    count = 0
    for _ in range(1000):
        upper = result.upper()
        lower = result.lower()
        parts = result.split("5")
        count += len(parts)
    print(f"operations: {count}")

if __name__ == "__main__":
    main()
